### PR TITLE
Add more tests and setup CodeClimate

### PR DESCRIPTION
**Pull request by szeck87**
* Add more tests to cover more code
* Enable CodeClimate

### CODE REVIEW

**Comment by szeck87** [comment]
Is there a way to make https://github.com/Unibeautify/unibeautify-cli/blob/ebd433bafba8b3071a8500b43de17023f29c7c72/src/Runner.ts#L111-L127 shorter?  The only difference between having a config file and not is whether to use `load` or `search` methods.

**Comment by Glavin001** [comment]
@szeck87 Something like:

```typescript
  private configFile(configFile?: string, filePath?: string) {
    const configExplorer = cosmiconfig("unibeautify", {});
    const promise = configFile ? configExplorer.load(configFile) : configExplorer.search(filePath);
    return promise.then(result => (result ? result.config : null))
      .catch(error =>
        Promise.reject(new Error(`Could not load configuration file ${configFile}`))
      );
  }
```

File: test/.unibeautifyrc.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (Glavin001) [inline test/.unibeautifyrc.yml]: @szeck87 `.unibeautifyrc` config file only works in the project root directory right now.

Comment (szeck87) [reply test/.unibeautifyrc.yml]: For the CI you mean?

Comment (Glavin001) [reply test/.unibeautifyrc.yml]: Correct. You're right, it does work properly using the editor extensions though.

Consider this a Unibeautify CI bug (or feature request) and ignore my comment above 😄 .

File: src/Runner.ts [hunk 2]
```diff
@@ -11,52 +11,57 @@ import * as cosmiconfig from "cosmiconfig";
...
+    } else {
+      return this.configFile(configFile);
+    }
+  }
+
+  private async readText(filePath?: string): Promise<string> {
+    if (this.isTerminal && filePath) {
+      return await this.readFile(filePath);
+    } else {
+      return await this.readFromStdin();
```
Comment (Glavin001) [inline src/Runner.ts]: `aync` and `await` for this function really is not necessary.

File: src/Runner.ts [hunk 3]
```diff
@@ -104,20 +110,16 @@ export class Runner {
...
-      return configExplorer
-        .load(configFile)
-        .then(result => (result ? result.config : null))
-        .catch(error => {
-          Promise.reject(error);
-        });
-    }
-    return configExplorer
-      .search(filePath)
+    const promise = configFile
```
Comment (Glavin001) [inline src/Runner.ts]: Should probably rename `promise` to something more meaningful. Even `loadConfigPromise` or something.

File: test/.unibeautifyrc.yml (hunk 1 above)
Comment (szeck87) [reply test/.unibeautifyrc.yml]: Yea this file was just meant to be used in the test cases, nothing else.

**Pull request merged**